src/tags_int.cpp

```diff
@@ -2767,15 +2767,17 @@ namespace Exiv2 {
             }
             std::ostringstream oss;
             oss.copyfmt(os);
-            const float sec = 3600 * value.toFloat(0)
-                              + 60 * value.toFloat(1)
-                              + value.toFloat(2);
+            const double t = 3600 * value.toFloat(0)
+                             + 60 * value.toFloat(1)
+                             + value.toFloat(2);
             int p = 0;
-            if (sec != static_cast<int>(sec)) p = 1;
-
-            const int hh = static_cast<int>(sec / 3600);
-            const int mm = static_cast<int>((sec - 3600 * hh) / 60);
-            const float ss = sec - 3600 * hh - 60 * mm;
+            const double fraction = fmod(t,1);
+            if (fraction != 0) p = 1;
+            const double seconds = t - fraction;
+            const double ss = fmod(seconds, 60);
+            const double minutes = (seconds - ss)/60;
+            const double mm = fmod(minutes, 60);
```

Comment:
I realize these will be printed as integers below, but maybe the code reflects the intent better if the cast to int for mm and hh is kept here?

Reply:
I updated it. I added an extra `enforce`, because the cast to `int` could crash if `t == NaN`. I also added an `fmod(hours,24)` which wasn't there in the old code, because otherwise casting the hours to `int` could overflow.